platform/api/equinixmetal/api.go

```diff
@@ -95,27 +96,63 @@ type Options struct {
 	InstallerImageCpioURL string
 	// e.g. https://alpha.release.flatcar-linux.net/amd64-usr/current/flatcar_production_packet_image.bin.bz2
 	ImageURL string
+	// Metro is where you want your server to live.
+	Metro string
 
-	// Options for Google Storage
-	GSOptions *gcloud.Options
 	// Google Storage base URL for temporary uploads
 	// e.g. gs://users.developer.core-os.net/bovik/mantle
 	StorageURL string
-	// Metro is where you want your server to live.
-	Metro string
+
+	// Options for Google Storage
+	GSOptions *gcloud.Options
 }
 
 type API struct {
-	c      *packngo.Client
-	bucket *storage.Bucket
-	opts   *Options
+	c       *packngo.Client
+	storage Storage
+	opts    *Options
 }
 
 type Console interface {
 	io.WriteCloser
 	SSHClient(ip, user string) (*ssh.Client, error)
 }
 
+// Storage defines a medium used to store the data temporary
+// created for a test.
+type Storage interface {
+	// Upload the []byte data to the implemented storage.
+	Upload(string, string, []byte) (string, string, error)
```

Comment:
This could use a bit of doc about what the string parameters and returns are expected to mean.

Reply:
Or can one add names?